### PR TITLE
Fix column format comparisons in io.fits tests

### PR DESCRIPTION
**Pull request by saimn**
Ref #14920

### CODE REVIEW

**Review by pllim — APPROVED** [review]
Thanks!

**Review by eerovaher — DISMISSED** [review]
In #14919 the [rule B015](https://beta.ruff.rs/docs/rules/#flake8-bugbear-b) was removed from the global Ruff ignore list and was added to the `io` sub-package ignore list because of the missing asserts that this pull request addresses, so `.ruff.roml` also needs to be edited so that Ruff will enforce B015 in `io`: https://github.com/astropy/astropy/blob/0a71105fbaa71439206d496a44df11abc0e3ac96/.ruff.toml#L291-L292

**Comment by eerovaher** [comment]
@saimn, are you planning to update the Ruff configuration any time soon or would you prefer me to do it?

File: astropy/io/fits/tests/test_table.py [hunk 1]
```diff
@@ -3377,12 +3377,14 @@ def test_column_format_interpretation(self):
         """
 
         for recformat, fitsformat in NUMPY2FITS.items():
+            if recformat == "a":
+                fitsformat = "0A"
```
Comment (nstarman) [inline astropy/io/fits/tests/test_table.py:3381]: Looking at this change, it's apparently problematic to edit ``NUMPY2FITS`` to change it to ``{"a":  "0A"}``.
Part of the problem appears to be that ``np.dtype("a") `` won't get the correct ``.format`` to compare to `fitsformat`. Am I understanding the problem correctly?

**Comment by eerovaher** [comment]
I went ahead and updated this pull request myself. I rebased to current `main` and edited Ruff configuration as required, but the changes to `astropy/io/fits/tests/test_table.py` are the same that @saimn had made when the pull request was opened.

**Review by astrofrog — APPROVED** [review]

**Comment by pllim** [comment]
Thanks, all!

**Pull request merged**